src/public/videoApp.ts

```diff
@@ -0,0 +1,146 @@
+   * Video frame format enum, currentlyl only support NV12
+   */
+  export enum VideoFrameFormat {
+    NV12,
+  }
+  /**
+   * Video frame configuration supplied to Teams to customize the generated video frame parameters, like format.
+   */
+  export interface VideoFrameConfig {
+    /**
+     * video format
+     */
+    format: VideoFrameFormat;
+  }
+
+  /**
+   *  Video effect change type enum
+   */
+  export enum EffectChangeType {
+    /**
+     * current video effect changed.
+     */
+    EffectChanged,
+    /**
+     * disable the video effect
+     */
+    EffectDisabled,
+  }
+
+  /**
+   *  Video frame call back funtion definition
+   */
+  type VideoFrameCallback = (
+    frame: VideoFrame,
+    notifyVideoFrameProcessed: () => void,
+    notifyError: (errorMessage: string) => void,
+  ) => void;
+
+  /**
+   *  Video effect change call back funtion definition
+   */
+  type VideoEffectCallBack = (effectId: string | undefined) => void;
+
+  /**
+   * VideoApp private members
+   */
+  class VideoAppPrivate {
+    public static videoFrameCallback: VideoFrameCallback;
+    public static videoEffectCallback: VideoEffectCallBack;
+  }
+  /**
+   * register to read the video frames in Permissions section.
+   */
+  export function registerForVideoFrame(frameCallback: VideoFrameCallback, config: VideoFrameConfig): void {
+    ensureInitialized(FrameContexts.sidePanel);
+    VideoAppPrivate.videoFrameCallback = frameCallback;
+    registerHandler('videoApp.newVideoFrame', (videoFrame: VideoFrame) => {
+      if (VideoAppPrivate.videoFrameCallback !== null && videoFrame !== undefined) {
+        VideoAppPrivate.videoFrameCallback(videoFrame, notifyVideoFrameProcessed, notifyError);
+      }
+    });
+    sendMessageToParent('videoApp.registerForVideoFrame', [config]);
+  }
+
+  /**
+   * VideoApp extension should call this to notify Teams Client current selected effect parameter changed.
+   * If it's pre-meeting, Teams client will call videoEffectCallback immediately then use the videoEffect.
+   * in-meeting scenario, we will call videoEffectCallback when apply button clicked.
+   * @param effectChangeType the effect change type.
+   * @param effectId Newly selected effect id.
+   */
+  export function notifySelectedVideoEffectChanged(
+    effectChangeType: EffectChangeType,
+    effectId: string | undefined,
+  ): void {
+    ensureInitialized(FrameContexts.sidePanel);
+    sendMessageToParent('videoApp.videoEffectChanged', [effectChangeType, effectId]);
+  }
+
+  /**
+   * Register the video effect callback, Teams client uses this to notify the videoApp extension the new video effect will by applied.
+   */
+  export function registerForVideoEffect(callback: VideoEffectCallBack): void {
+    ensureInitialized(FrameContexts.sidePanel);
+    VideoAppPrivate.videoEffectCallback = callback;
+    registerHandler('videoApp.effectParameterChange', (effectId: string) => {
+      if (VideoAppPrivate.videoEffectCallback !== undefined) {
```

Comment:
register Handler already register handle inside. so this function should be something like

export function registerForVideoEffect(callback: VideoEffectCallBack): void {
      
 ensureInitialized(FrameContexts.sidePanel); 
 registerHandler('videoApp.effectParameterChange', callback);
}

Reply:
Thanks for pointing it out. Will fix it.